### ISSUE TITLE
Infinite recursion BitType operator==

### ISSUE THREAD

**Issue by timo-suoranta-varjo** [KhronosGroup/Vulkan-Hpp#752]
I am running into infinite loop with
`vk::QueueFlags flags; ... bool graphics_supported = (flags & vk::QueueFlagBits::eGraphics) == vk::QueueFlagBits::eGraphics;`

Apparently
```
  template <typename BitType>
  VULKAN_HPP_CONSTEXPR bool operator==(BitType bit, Flags<BitType> const& flags) VULKAN_HPP_NOEXCEPT
  {
    return flags == bit;
  }
```
keeps calling itself recursively.

This happens with clang (version 10.0).
This does not happen with gcc (version 7.5).

**Comment by asuessenbach** [comment]
Infinite loop sounds like a runtime issue.
Do you get any warnings on that line of code?
Are you using any different settings between gcc 7.5 and gcc 10.0?
Against what C++ Language Standard are you compiling (C++11/C++14/C++17/C++20) ?
Can you check if VULKAN_HPP_HAS_SPACESHIP_OPERATOR gets defined in vulkan.hpp?

**Comment by timo-suoranta-varjo** [comment]
I verified the behavior by placing breakpoint into the line which calls the operator and stepping the code in debugger to see that the operator calls itself.
I am using all of vulkan loader, validator, vulkanhpp built from sources, latest from github repositories.
I am compiling against C++17, so no spaceship operator. Settings should be similar; I use CMake and only choose different kit.

**Comment by asuessenbach** [comment]
Unfortunately, we can't repro that issue!
But could you please modify your local vulkan.hpp a little?
In that operator==() mentioned above, please replace
    `return flags == bit;`
with
    `flags.operator==(bit)`
Does that prevent that infinite recursion?

**Comment by asuessenbach** [comment]
Should be resolved by #755, closing.
